src/Core/HTTPApplication.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace SilverStripe\Core;
+
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+
+/**
+ * Invokes the HTTP application within an ErrorControlChain
+ */
+class HTTPApplication implements Application
```

Comment:
I wonder if this should be part of SilverStripe\Control, since ti depends on HTTPRequest response?